### PR TITLE
Allow --commcare-rev with --set on deploy

### PR DESCRIPTION
**Pull request by dannyroberts**
Previously if you use --set, it will effectively overwrite the value
of --commcare-rev (since that also uses --set on fab downstream)

This change also removes some of the awkwardness of using --set
where you have to then explictly pass in the component arg or it
treats the value of --set as the component arg (and fails).
Now there is none of that.

### CODE REVIEW

File: src/commcare_cloud/commands/deploy.py [hunk 1]
```diff
@@ -31,11 +31,14 @@ class Deploy(CommandBase):
...
         """, default=None),
+        Argument('--set', dest='fab_settings', help="""
+            fab settings in k1=v1,k2=v2 format to be passed down to fab 
+        """, default=None),
         shared_args.QUIET_ARG,
         shared_args.BRANCH_ARG,
     )
 
-    def run(self, args, unknown_args):
+    def run(self, args, unknown_args, color_warn=None):
```
Comment (orangejenny) [inline src/commcare_cloud/commands/deploy.py:41]: How is `color_warn` supplied? It looks like lines 57-61 will fail hard if it's `None`.

File: src/commcare_cloud/commands/deploy.py [hunk 2]
```diff
@@ -50,15 +53,23 @@ def run(self, args, unknown_args):
                 if deploy_component != 'both':
                     _warn_no_formplayer()
                 self.deploy_commcare(environment, commcare_rev, args, unknown_args)
-
         if deploy_component in ['formplayer', 'both']:
+            if deploy_component != 'both':
```
Comment (orangejenny) [inline src/commcare_cloud/commands/deploy.py:57]: Nitpick: this would be clearer if it were `if deploy_component == 'formplayer'`

Comment (dannyroberts) [reply src/commcare_cloud/commands/deploy.py:57]: I was following the convention three/four lines above 🤷

File: src/commcare_cloud/commands/deploy.py (hunk 1 above)
Comment (dannyroberts) [reply src/commcare_cloud/commands/deploy.py]: Thanks this was totally me not understanding what my IDE was doing for me, fixed now

File: src/commcare_cloud/commands/deploy.py (hunk 2 above)
Comment (orangejenny) [reply src/commcare_cloud/commands/deploy.py:57]: fair enough

**Pull request merged**